lib/imagickal.js

```diff
@@ -100,45 +106,51 @@ exports.identify = function (path, verifyImage, callback) {
 		verifyImage = false;
 	}
 
-	const cmd = printf(
-		'identify -format "{\\"format\\":\\"%m\\",\\"width\\":%w,\\"height\\":%h}" %s%s',
-		verifyImage ? '-verbose ' : '',
-		path instanceof Readable ? '-' : path
-	);
+	const input_path = path instanceof Readable ? '-' : path;
+	var validFilename = /^[0-9a-zA-Z._@()-]+$/i.test(input_path);
+	if (validFilename === true) {
+		const cmd = printf(
+			'identify -format "{\\"format\\":\\"%m\\",\\"width\\":%w,\\"height\\":%h}" %s%s',
+			verifyImage ? '-verbose ' : '',
+			input_path
+		);
 
-	debug('identify', cmd);
+		debug('identify', cmd);
 
-	const promise = new Promise((resolve, reject) => {
+		const promise = new Promise((resolve, reject) => {
 		const shell = exec(cmd, Object.assign({ stdio: ['pipe'] }, globalOpts.execOptions || {}), (err, stdout) => {
-			if (err) {
-				if (err.message.match(/decode delegate/)) {
-					return reject(new Error('Invalid image file'));
+				if (err) {
+					if (err.message.match(/decode delegate/)) {
+						return reject(new Error('Invalid image file'));
+					}
+					return reject(err);
 				}
-				return reject(err);
-			}
 
-			let data;
-			try {
-				data = parseOutput(stdout.toString());
-				data.format = data.format.toLowerCase();
-				data.format = data.format === 'jpeg' ? 'jpg' : data.format;
-			} catch (e) {
-				return reject(new Error('Unable to parse identify data, output was:' + stdout.toString()));
-			}
+				let data;
+				try {
+					data = parseOutput(stdout.toString());
+					data.format = data.format.toLowerCase();
+					data.format = data.format === 'jpeg' ? 'jpg' : data.format;
+				} catch (e) {
+					return reject(new Error('Unable to parse identify data, output was:' + stdout.toString()));
+				}
 
-			if (!data) {
-				return reject(new Error('Unable to identify image, output was:' + stdout.toString()));
-			}
+				if (!data) {
+					return reject(new Error('Unable to identify image, output was:' + stdout.toString()));
+				}
 
-			resolve(data);
-		});
+				resolve(data);
+			});
 
-		if (path instanceof Readable) {
-			path.pipe(shell.stdin);
-		}
-	});
+			if (path instanceof Readable) {
+				path.pipe(shell.stdin);
+			}
+		});
 
-	return promise.asCallback(callback);
+		return promise.asCallback(callback);
+	} else {
+		console.log("Path validation failed");
```

Comment:
remove console.log()

Reply:
Wouldn't a developer/user need it when running it in debug mode?